clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetcherTest.java

```diff
@@ -148,6 +150,33 @@ public boolean matches(ClientRequest request) {
     }
 
     @Test
+    public void testFetchedRecordsRaisesOnSerializationErrors() {
+        ByteArrayDeserializer deserializer = new ByteArrayDeserializer() {
+            @Override
+            public byte[] deserialize(String topic, byte[] data) {
+                throw new SerializationException();
```

Comment:
Yeah, that works. I was even thinking of just extending SerializationException with that information and leaving the existing constructors. But I don't think it matters much whether it's a new ConsumerSerializationException or (the slightly awkward) SerializationException wrapping another SerializationException with the info just placed directly in the message.